packages/lib/messaging/src/MessageProxy.ts

```diff
@@ -0,0 +1,60 @@
+import {
+    DeliveryServiceProfile,
+    GetResource,
+    getDeliveryServiceProfile,
+    getUserProfile,
+} from 'dm3-lib-profile';
+import { ProxyEnvelop } from './ProxyEnvelop';
+import { ethers } from 'ethers';
+import { log } from 'dm3-lib-shared';
+import { EncryptionEnvelop } from './Envelop';
+
+export async function sendOverMessageProxy(
+    provider: ethers.providers.JsonRpcProvider,
+    proxyEnvelop: ProxyEnvelop,
+    getRessource: GetResource<DeliveryServiceProfile>,
+    submitMessage: (url: string, envelop: EncryptionEnvelop) => Promise<void>,
+) {
+    const userProfile = await getUserProfile(provider, proxyEnvelop.to);
+
+    if (!userProfile) {
+        throw Error(`Couldn't fetch user profile`);
+    }
+
+    let success = false;
+    for (const dsName of userProfile.profile.deliveryServices) {
+        try {
+            const dsProfile = await getDeliveryServiceProfile(
+                dsName,
+                provider,
+                getRessource,
+            );
+
+            if (dsProfile) {
+                const envelopContainer = proxyEnvelop.encryptionEnvelops.find(
+                    (envelops) => envelops.deliveryServiceEnsName === dsName,
+                );
+
+                if (!envelopContainer) {
+                    throw Error(`Couldn't find an envelop for ${dsName}`);
+                }
+
+                await submitMessage(dsProfile.url, {
+                    ...envelopContainer.encryptionEnvelop,
+                    message: proxyEnvelop.encryptedMessage,
+                });
+                success = true;
+                return;
+            }
+        } catch (e) {
+            log(
+                `Failed to send message using: ${dsName} (${
+                    (e as Error).message
+                })`,
+            );
+        }
+    }
+    if (!success) {
```

Comment:
I don't think you need that if statement because the function never will reach this part of the function when at least one promise is returned successfully. 
Hence it might be sufficient just to throw here and remove the "success" var.